### DISCUSSION TITLE
Would it be possible to move from jsonschema 2.6 to 3.*

### DISCUSSION THREAD

**Discussion by james-bellamy** [StackStorm/st2#5696] [Q&A]
Hello, I hope this is the right place for this question. 

We're attempting to integrate st2client into our project and it has a strict requirement of jsonschema 2.6 which is unfortunately incompatible with our system. Could the version constraints be loosened? 

It appears to be set in https://github.com/StackStorm/orquesta/blob/676ce3e36b2663ba6c33ea9837d61d3e55a12981/requirements.txt#L4 and  https://github.com/StackStorm/st2/blob/9bab2a381a9d91a6571cab7cfcef9c96225d5f36/fixed-requirements.txt#L22 however the only use I'm seeing is to validate against draft 3, draft 4, and a customer validator based on draft 4. According to https://github.com/python-jsonschema/jsonschema#features the most recent version, and presumably all versions since 2.6, support draft 3 and draft 4.

**Comment by amanda11** [comment]
We would need to use a version that can be used on python3.6->3.9, as at the moment the same version has to be used for all our distributions, and I think there isn't a 4.x that works on python3.6. (When I attempt the latest I can get is 4.0.0 which has been yanked).
But version 3.2.0 might be a potential candidate.
It's possible the restriction is still in place from when we had to support python 2.7, but that is no-longer a  request.

We welcome fixes provided by the community, so if you were interested in contributing a PR to see if the version can be incremented to 3.2.0 then that would be great. There is mention in 3.0 release of a major overhaul to some of the classes, so it is possible that there may be additional code changes required to support moving up to 3.2. If you are interested in investigating further, then feel free to start looking at PRs on those two repos, and if you reach out to the generic #development channel on slack, I'm sure we can assist.
